trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/security/S3DatabaseSecurityFacade.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.aws.proxy.spi.security;
+
+import java.util.Optional;
+
+public interface S3DatabaseSecurityFacade
+{
+    S3DatabaseSecurityFacade DEFAULT = new S3DatabaseSecurityFacade() {};
+
+    default Optional<String> tableName(String bucketName, Optional<String> key)
+    {
+        return Optional.empty();
+    }
+
+    default SecurityResponse nonTableOperation(String httpVerb, String bucketName, Optional<String> lowercaseAction)
```

Comment:
I _think_ both this and `tableOperation` should also get access to query parameters and headers.

The first use-case I have in mind is [listing object in a bucket](https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html), where we probably want to include the `prefix` query parameter in the access control decision.

Another operation is `CopyObject`, is a PUT request with no action. A request is essential a `CopyObject` if it has the `x-amz-copy-source` header.

Reply:
This is the benefit of `S3DatabaseSecurityFacadeProvider`. `S3DatabaseSecurityFacade` is provided by `S3DatabaseSecurityFacadeProvider` which gets a `ParsedS3Request` argument. So, implementations of `S3DatabaseSecurityFacade` that need request values can get them that way.

Reply:
Reading the above, I realize the `verb` isn't needed. I'll remove it.